### PR TITLE
H-1620: Move `api` out of `graph` into `graph-api`

### PR DESCRIPTION
**Pull request by indietyp**
## 🌟 What is the purpose of this PR?

This moves the HASH API module into a separate crate. This has many advantages, such as a higher degree of separation and the removal of the `build.rs` from the main graph crate. In the future one might also want to move out the logging logic or the database connection logic into separate crates.

### 🚢 Has this modified a publishable library?

This PR:

- [x] does not modify any publishable blocks or libraries, or modifications do not need publishing

### 📜 Does this require a change to the docs?

The changes in this PR:

- [x] are internal and do not require a docs change

### 🕸️ Does this require a change to the Turbo Graph?

The changes in this PR:

- [x] I am unsure / need advice

## ⚠️ Known issues

* CI needs to be successful
* some things in the graph still use `utoipa`. That's bad. Either feature flag or try to move it into API

## 🐾 Next steps

* alternative `tarpc` based implementation

## 🛡 What tests cover this?

Existing test suite

## ❓ How to test this?

Look at CI

### CODE REVIEW

File: apps/hash-graph/libs/api/src/rest/api_resource.rs [hunk 1]
```diff
@@ -1,15 +1,16 @@
...
-use crate::{api::rest::RestApiStore, store::StorePool};
+use crate::rest::RestApiStore;
 
 /// With REST, we define resources that can be acted on. These resources are defined through
 /// routes and HTTP methods.
 ///
 /// This trait encapsulates a way to define related resource operations
 /// through a `Router`, making it explicitly clear we want to provide `OpenApi` specification as
 /// documentation for the routes.
-pub(super) trait RoutedResource: utoipa::OpenApi {
```
Comment (indietyp) [inline apps/hash-graph/libs/api/src/rest/api_resource.rs:12]: Looking through this, `RustRover` was a bit... ambitioned while moving.

File: apps/hash-graph/libs/api/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,46 @@
+[package]
+name = "graph-api"
+version = "0.1.0"
+authors.workspace = true
+edition.workspace = true
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
```
Comment (TimDiekmann) [inline apps/hash-graph/libs/api/Cargo.toml:8]:
```suggestion
# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

```

File: apps/hash-graph/libs/api/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+tokio-util = { workspace = true, features = ["io"], optional = true }
+tower = "0.4.13"
+tower-http = { version = "0.5.0", features = ["trace"] }
+tracing = "0.1.40"
+tracing-opentelemetry = "0.22.0"
+utoipa = "4.1.0"
+uuid = "1.6.1"
+
+[features]
+test-server = ["dep:tokio", "dep:tokio-postgres", "dep:tokio-util", "dep:codec"]
```
Comment (TimDiekmann) [inline apps/hash-graph/libs/api/Cargo.toml:46]: Interesting that these dependencies are only required for the test server 👀

**Review by TimDiekmann — APPROVED** [review]
Great, I wanted to do that a long time ago, thank you! ✨ 🚀 

One non-blocking comment

File: apps/hash-graph/libs/api/Cargo.toml (hunk 3 above)
Comment (indietyp) [reply apps/hash-graph/libs/api/Cargo.toml:46]: Yep I was surprised myself 👀

**Review by TimDiekmann — DISMISSED** [review]

File: apps/hash-graph/libs/api/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,44 @@
+[package]
+name = "graph-api"
+version = "0.1.0"
+authors.workspace = true
+edition.workspace = true
```
Comment (TimDiekmann) [inline apps/hash-graph/libs/api/Cargo.toml:5]:
```suggestion
[package]
name = "graph-api"
version = "0.0.0"
authors.workspace = true
edition.workspace = true
publish = false
description = "HASH Graph API"
```

**Review by TimDiekmann — APPROVED** [review]

**Review by TimDiekmann — APPROVED** [review]

**Pull request merged**